src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -227,25 +228,97 @@ private async Task UpdateJobRecordAsync(CancellationToken cancellationToken)
             EnsureArg.IsNotNull(progress, nameof(progress));
             EnsureArg.IsNotNull(sharedQueryParametersList, nameof(sharedQueryParametersList));
 
-            // Current batch will be used to organize a set of search results into a group so that they can be committed together.
-            string currentBatchId = progress.Page.ToString("d6");
-
             List<Tuple<string, string>> queryParametersList = new List<Tuple<string, string>>(sharedQueryParametersList);
             if (progress.ContinuationToken != null)
             {
                 queryParametersList.Add(Tuple.Create(KnownQueryParameterNames.ContinuationToken, progress.ContinuationToken));
             }
 
-            if (_exportJobRecord.ExportType == ExportJobType.Patient)
+            var requestedResourceTypes = _exportJobRecord.ResourceType?.Split(',');
+            var filteredResources = new HashSet<string>();
+            if (_exportJobRecord.Filters != null)
             {
-                queryParametersList.Add(Tuple.Create(KnownQueryParameterNames.Type, KnownResourceTypes.Patient));
+                foreach (var filter in _exportJobRecord.Filters)
+                {
+                    filteredResources.Add(filter.ResourceType.ToUpper(CultureInfo.InvariantCulture));
```

Comment:
>ToUpper [](start = 62, length = 7)

Curious why we are doing ToUpper() here

Reply:
It was to avoid issues with users entering the resource type in different cases. I would use StringComparison but the List Contains method doesn't allow you to pass one so I need to convert them to a consistent case.

Reply:
There is an option to set the comparer when instantiating the HashSet. Using that we should be able to avoid the `ToUpper()` conversions `var filteredResources = new HashSet<string>(StringComparer.OrdinalIgnoreCase)`

---
In reply to: [537724269](https://github.com/microsoft/fhir-server/pull/1462#discussion_r537724269) [](ancestors = 537724269)